### PR TITLE
(GH-actions) add `allow-licenses` list in dependency_review #PR2

### PR DESCRIPTION
**Pull request by kvramyasri7**
#### Description of changes

add `allow-licenses` list in dependency_review.

Used them as a source of truth: 

- https://opensource.org/licenses/ 
- https://spdx.org/licenses/

for checking the list of licenses.

#### Description of how you validated changes
https://github.com/aws-amplify/amplify-ui/pull/4145 Took this as reference PR from sibling team.

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -44,7 +41,6 @@ jobs:
     needs:
       - unit-tests
       - bundle-size-tests
-      - license-test
```
Comment (ashwinkumar6) [inline .github/workflows/pr.yml:47]: I think this servers a different purpose. This workflow in turn calls `yarn test:license` which checks if new files which are added have our [AWS license header](https://github.com/aws-amplify/amplify-js/pull/11590/files#diff-9e861f5fb09be89278f63b38eec623c20a798e12255eed73226e612d91ad186e).
[PR](https://github.com/aws-amplify/amplify-js/pull/11590) where we added this feature

On the other hand the [actions/dependency-review-action](https://github.com/actions/dependency-review-action) checks if new dependencies introduced have the required license so we don't add new packages with restricted licenses 

I feel we can improve the naming of the workflow though, I think @jimblanc might also have some context

**Review by ashwinkumar6 — COMMENTED** [review]
Thanks for adding this in @kvramyasri7 
Added a comment regarding removing existing `license-test` workflow 

I think it'll be great to also get an approval from product for the acceptable licenses listed in `dependecy-review-config.yml` just in case

File: .github/workflows/pr.yml (hunk 1 above)
Comment (kvramyasri7) [reply .github/workflows/pr.yml:47]: Added this part. Thanks @ashwinkumar6.

File: .github/dependency-review/dependecy-review-config.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - 'PostgreSQL'
+  - 'Python-2.0'
+  - 'Spencer-94'
+  - 'Unicode-DFS-2015'
+  - 'Unicode-DFS-2016'
+  - 'Unlicense'
+  - 'WTFPL'
+  - 'X11'
+  - 'zlib-acknowledgement'
+  - 'Zlib'
```
Comment (elorzafe) [inline .github/dependency-review/dependecy-review-config.yml:34]: any reason to not what currently UI team has https://github.com/aws-amplify/amplify-ui/blob/main/.github/dependency-review/config.yml ?

Comment (kvramyasri7) [reply .github/dependency-review/dependecy-review-config.yml:34]: from the conversation I had with one of the engineer from ui team, this list is subjected to change. Referencing the latest guide from oss is the tip I had.

Comment (kvramyasri7) [reply .github/dependency-review/dependecy-review-config.yml]: I have updated the license list to Amazon `pre-approved` list  @elorzafe. Please take a look at it now. If you find something that should not be in the list let me know we can chat with the people who made the list and communicate with ui-team as well.

File: .github/dependency-review/dependecy-review-config.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+allow-licenses:
+  - '0BSD'
```
Comment (elorzafe) [inline .github/dependency-review/dependecy-review-config.yml:2]: I couldnt find this on the pre approved list

**Comment by elorzafe** [comment]
@ovalba I would like your feedback on this. Thanks!

**Review by elorzafe — APPROVED** [review]
Thanks @kvramyasri7 !

**Review by kvramyasri7 — COMMENTED** [review]
@calebpollman  `amplify-ui` code base can be at risk. Please take a look at it when you get a chance.

**Review by ashwinkumar6 — APPROVED** [review]

**Pull request merged**